tests/click_tests/test_reset_slip39_advanced.py

```diff
@@ -15,164 +15,113 @@
 from . import reset
 
 if TYPE_CHECKING:
     from ..device_handler import BackgroundDeviceHandler
 
 
-EXTERNAL_ENTROPY = b"zlutoucky kun upel divoke ody" * 2
+pytestmark = [pytest.mark.skip_t1]
 
-with_mock_urandom = mock.patch("os.urandom", mock.Mock(return_value=EXTERNAL_ENTROPY))
 
-
-@pytest.mark.skip_t1
 @pytest.mark.setup_client(uninitialized=True)
-@with_mock_urandom
 def test_reset_slip39_advanced_2of2groups_2of2shares(
     device_handler: "BackgroundDeviceHandler",
 ):
-    features = device_handler.features()
-    debug = device_handler.debuglink()
-
-    assert features.initialized is False
-
-    device_handler.run(
-        device.reset,
-        backup_type=messages.BackupType.Slip39_Advanced,
-        pin_protection=False,
-    )
-
-    # confirm new wallet
-    reset.confirm_wait(debug, "Wallet creation")
-
-    # confirm back up
-    reset.confirm_read(debug, "Success")
-
-    # confirm checklist
-    reset.confirm_read(debug, "Checklist")
-
-    # set num of groups
-    reset.set_selection(debug, buttons.RESET_MINUS, 3)
-
-    # confirm checklist
-    reset.confirm_read(debug, "Checklist")
-
-    # set group threshold
-    reset.set_selection(debug, buttons.RESET_MINUS, 0)
-
-    # confirm checklist
-    reset.confirm_read(debug, "Checklist")
-
-    # set share num and threshold for groups
-    for _ in range(2):
-        # set num of shares
-        reset.set_selection(debug, buttons.RESET_MINUS, 3)
-
-        # set share threshold
-        reset.set_selection(debug, buttons.RESET_MINUS, 0)
-
-    # confirm backup warning
-    reset.confirm_read(debug, "Caution")
-
-    all_words: list[str] = []
-    for _ in range(2):
-        for _ in range(2):
-            # read words
-            words = reset.read_words(debug, True)
+    _slip39_advanced_reset(device_handler, 2, 2, 2, 2)
 
-            # confirm words
-            reset.confirm_words(debug, words)
-
-            # confirm share checked
-            reset.confirm_read(debug, "Success")
-
-            all_words.append(" ".join(words))
-
-    # confirm backup done
-    reset.confirm_read(debug, "Success")
-
-    # generate secret locally
-    internal_entropy = debug.state().reset_entropy
-    assert internal_entropy is not None
-    secret = generate_entropy(128, internal_entropy, EXTERNAL_ENTROPY)
-
-    # validate that all combinations will result in the correct master secret
-    reset.validate_mnemonics(all_words, secret)
 
-    assert device_handler.result() == "Initialized"
-
-    features = device_handler.features()
-    assert features.initialized is True
-    assert features.needs_backup is False
-    assert features.pin_protection is False
-    assert features.passphrase_protection is False
-    assert features.backup_type is messages.BackupType.Slip39_Advanced
-
-
-@pytest.mark.skip_t1
 @pytest.mark.setup_client(uninitialized=True)
 @pytest.mark.slow
-@with_mock_urandom
 def test_reset_slip39_advanced_16of16groups_16of16shares(
```

Comment:
huh so the cases were identical except for the parameters? let's parametrize the single one then (we can add marks to individual parameters via `pytest.param` to indicate one is slow and the other is not)

Reply:
> Would be still possible with parametrization, but current possibility of `-k 1of1` is easier

`pytest.param(..., id="1of1")` ?

Reply:
Done in [b0522bb](https://github.com/trezor/trezor-firmware/pull/2803/commits/b0522bbd7ee7a16be1fce9bf3c38f360f6879bdd)